packages/oasis-actions/src/actions/common.ts

```diff
@@ -81,21 +81,46 @@ export function sendToken(args: { asset: string; to: string; amount: BigNumber }
 }
 
 export function takeAFlashLoan(args: {
+  flashloanType: FLASHLOAN_TYPE
   flashloanAmount: BigNumber
   borrower: string
   dsProxyFlashloan: boolean
   calls: ActionCall[]
 }) {
-  return createAction(
-    getActionHash(CONTRACT_NAMES.common.TAKE_A_FLASHLOAN),
-    [calldataTypes.common.TakeAFlashLoan],
-    [
-      {
+
+  let actionName;
+  let flashloanCalldataType;
+  let flashloanCalldata;
+
+  switch(args.flashloanType) {
+    case FLASHLOAN_TYPE.FMM:
+      actionName = CONTRACT_NAMES.common.TAKE_A_FLASHLOAN
+      flashloanCalldataType = calldataTypes.common.TakeAFlashLoan
+      flashloanCalldata = {
+        amount: args.flashloanAmount.toFixed(0),
+        dsProxyFlashloan: args.dsProxyFlashloan,
+        calls: args.calls,
+      }
+      break
+    case FLASHLOAN_TYPE.BALANCER:
+      actionName = CONTRACT_NAMES.common.TAKE_A_BALANCER_FLASHLOAN
```

Comment:
I guess you can return the flashloan from the case block directly instead of assigniong values to some variables..